packages/token-bridge-sdk/src/util/index.ts

```diff
@@ -14,3 +19,105 @@ export const validateTokenList = (tokenList: TokenList) => {
 
   return validate(tokenList)
 }
+
+export function getDefaultTokenName(address: string) {
+  const lowercased = address.toLowerCase()
+  return (
+    lowercased.substring(0, 5) +
+    '...' +
+    lowercased.substring(lowercased.length - 3)
+  )
+}
+
+export function getDefaultTokenSymbol(address: string) {
+  const lowercased = address.toLowerCase()
+  return (
+    lowercased.substring(0, 5) +
+    '...' +
+    lowercased.substring(lowercased.length - 3)
+  )
+}
+
+/**
+ * Retrieves data about an ERC-20 token using its L1 address. Throws if fails to retrieve balance or allowance.
+ * @param erc20L1Address
+ * @returns
+ */
+export async function getL1TokenData(
+  account: string,
+  erc20L1Address: string,
+  l1Provider: JsonRpcProvider,
+  l2Provider: JsonRpcProvider,
+  throwOnInvalidERC20 = true
+): Promise<L1TokenData> {
+  const l2Network = await getL2Network(l2Provider)
+  const erc20Bridger = new Erc20Bridger(l2Network)
```

Comment:
could we pass in `erc20Bridger` instead of `l2Provider`?

Reply:
Let's not do that, we want to abstract away as much as possible of the SDK internals